kvmagent/kvmagent/plugins/vm_plugin.py

```
@@ -3220,19 +3261,23 @@ def loop_undefine(_):
 
             def force_undefine():
                 try:
-                    self.domain.undefine()
+                    flags = libvirt.VIR_DOMAIN_UNDEFINE_KEEP_NVRAM
+                    if tpm.VIRSH_SUPPORT_KEEP_TPM:
+                        flags |= VIR_DOMAIN_UNDEFINE_KEEP_TPM
+                    self.domain.undefineFlags(flags)
                 except:
```

Ruff (0.15.17)

[error] Do not use bare `except`

(E722)